nucypher/blockchain/eth/actors.py

```diff
@@ -666,6 +515,127 @@ def set_min_reward_rate_range(self,
         return receipt
 
 
+class Allocator:
+    def __init__(self, filepath: str, registry, deployer_address):
+
+        self.log = Logger("allocator")
+        self.staking_agent = ContractAgency.get_agent(StakingEscrowAgent,
+                                                      registry=registry)  # type: StakingEscrowAgent
+        self.max_substakes = self.staking_agent.contract.functions.MAX_SUB_STAKES().call()
+
+        with open(filepath, 'r') as allocation_file:
+            if filepath.endswith(".csv"):
+                reader = csv.DictReader(allocation_file)
+                allocation_data = list(reader)
+            else:  # Assume it's JSON by default
+                allocation_data = json.load(allocation_file)
+
+        self.allocations = dict()
+        self.deposited = set()
+        self.economics = EconomicsFactory.get_economics(registry)
+
+        total_to_allocate = 0
+
+        # Pre-process allocations data
+        for entry in allocation_data:
```

Comment:
There is not enough handling around this runtime boundary here - catching the `KeyError` and moving this to a processing method will be an improvement IMO.

Reply:
✅